samples/nrf9160/modem_shell/src/main.c

```diff
@@ -264,8 +246,11 @@ void main(void)
 		&cfg);
 
 #if !defined(CONFIG_LWM2M_CARRIER)
-	/* Get Modem library initialization return value. */
-	err = modem_lib_init_result;
+	/* Manually initialize modem library when CONFIG_LWM2M_CARRIER is disabled.
+	 * This is necessary for the modem trace flash backend. Which depend on the flash device
+	 * to be initialized before initializing the nRF modem library.
+	 */
```

Comment:
I think this comment could be removed, because manual initialization is now the only available option.